generator/src/test/java/com/linkedin/pegasus/generator/TestPegasusDataTemplateGenerator.java

```diff
@@ -205,4 +206,64 @@ private void testRunGenerator(String pegasusFilename, Map<String, String> expect
             file -> file.getName().replace(".java", ""),
             Function.identity()));
   }
+
+  /**
+   *
+   * @return an array of test cases where each case has two array of test schema file names. Those file names are
+   *   in the different permutations of same group of test schema files
+   */
+  @DataProvider(name = "test_schema_permutation_determinisim")
+  private Object[][] createPermutedDataTemplateCases()
+  {
+    return new Object[][]
+        {
+            {new String[]{"IsoDuration.pdsc", "PremiumService.pdsc"}, new String[]{"PremiumService.pdsc", "IsoDuration.pdsc"}},
+            {new String[]{"AField.pdl", "ARecord.pdl"}, new String[]{"ARecord.pdl", "AField.pdl"}},
+            {new String[]{"BRecord.pdl", "BField.pdl"}, new String[]{"BField.pdl", "BRecord.pdl"}},
+        };
+  }
+
+  @Test(dataProvider = "test_schema_permutation_determinisim")
+  public void testDataTemplateGenerationDeterminism(String[][] testArgs)
```

Comment:
I guess I made a mistake in my suggestion, what I wanted is for this method to take 2 args:
(String[] inputSchemasOrder1, String[] inputSchemasOrder2)

Reply:
I agree, is this not how this test framework organizes the inputs for DataProviders?

Reply:
This is addressed